File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-4.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int ans ;
        int st = 0;
        int end = nums.size()-1;

        while(st<=end){
            int mid = (st+end)/2;

            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]>=nums[0]){
                if(nums[mid]>target and nums[0]<=target){
                    end = mid-1;
                }

                else{
                    st = mid+1;
                }
            }
            else{
                if(nums[mid]<target and target< nums[0]){
                    st = mid+1;
                }
                else{
                    end = mid-1;
                }
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-4.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        start = 0
        end = len(nums)-1
        ans = 0
        
        while start<=end:
            mid = (end+start)//2
            if nums[0]>nums[mid]:
                end = mid-1
                ans = mid
            else:
                if nums[0]>nums[end]:
                    start = mid+1
                else:
                    end = mid-1

            print(start,end)
            
        return nums[ans]
